utils/utils.js

```diff
@@ -743,6 +749,77 @@ const getStringId = (id) => {
     : JSON.stringify(id);
 };
 
+/**
+ * A function to validate and return Residency server input
+ * @returns string/undefined
+ */
+const getResidencyServer = (options) => {
+  const region = options ? options.residencyServer : undefined;
+  if (region) {
+    if (typeof region !== 'string' || !RESIDENCY_SERVERS.includes(region.toUpperCase())) {
+      logger.error('Invalid residencyServer input');
+      return undefined;
+    }
+    return region.toUpperCase();
+  }
+  return undefined;
+};
+
+const isValidServerUrl = (serverUrl) => {
+  if (!serverUrl || typeof serverUrl !== 'string' || serverUrl.trim().length === 0) {
+    return false;
+  }
+  return true;
+};
+
+/**
+ * A function to get url from source config response
+ * @param {array} urls    An array of objects containing urls
+ * @returns
+ */
+const getDefaultUrlofRegion = (urls) => {
+  let url;
+  if (Array.isArray(urls) && urls.length) {
+    const obj = urls.find((elem) => elem.default === true);
+    if (obj && obj.url && isValidServerUrl(obj.url)) {
```

Comment:
Redundant.
```suggestion
    if (isValidServerUrl(obj.url)) {
```

Reply:
obj might be undefined, in that case, `obj` presence checking is needed omitted `obj.url` checking